sympy/matrices/matrices.py

```diff
@@ -1677,6 +1678,40 @@ def limit(self, *args):
 # https://github.com/sympy/sympy/pull/12854
 class MatrixDeprecated(MatrixCommon):
     """A class to house deprecated matrix methods."""
+    def _legacy_array_dot(self, b):
+        """Compatibility function for deprecated behavior of ``matrix.dot(vector)``
+        """
+        from .dense import Matrix
+
+        if not isinstance(b, MatrixBase):
+            if is_sequence(b):
+                if len(b) != self.cols and len(b) != self.rows:
+                    raise ShapeError(
+                        "Dimensions incorrect for dot product: %s, %s" % (
+                            self.shape, len(b)))
+                return self.dot(Matrix(b))
+            else:
+                raise TypeError(
+                    "`b` must be an ordered iterable or Matrix, not %s." %
+                    type(b))
+
+        mat = self
+        if mat.cols == b.rows:
+            if b.cols != 1:
+                mat = mat.T
+                b = b.T
+            prod = flatten((mat * b).tolist())
+            if len(prod) == 1:
```

Comment:
This will be handled by new `dot`.

Reply:
if `len(prod)` is 1 then it `mat` and `b` would have been row and column matrix so code will not go in `_legacy_array_dot()`.

Reply:
@vishalg2235 As you said, `len(prod) == 1` will always be false, so this check(L#1704-1705) is redundant.

Reply:
yes it is redundant. I thought while deprecation whole old code should be kept unchanged, But yes that check is just waste of lines in that function. I will remove it.